forge-game/src/main/java/forge/game/card/CardFactoryUtil.java

```diff
@@ -1255,6 +1225,22 @@ public static void addTriggerAbility(final KeywordInterface inst, final Card car
 
             saRebel.setIntrinsic(intrinsic);
             inst.addTrigger(etbTrigger);
+        } else if (keyword.equals("Gift")) {
+            // Gift is a special keyword that is used to create a trigger for permanents when they enter the battlefield
+            // On casting a Gift needs to be promised to an opponent
+            final SpellAbility saGift = AbilityFactory.getAbility(card.getSVar("GiftEffect"), card);
```

Comment:
What happens when you do this for non-Permanents?

`card.getFirstSpellAbility().setAdditionalAbility("GiftAbility", saGift)` ?